libraries/lib-project-file-io/SqliteSampleBlock.cpp

```diff
@@ -289,6 +298,35 @@ SampleBlockPtr SqliteSampleBlockFactory::DoCreateFromXML(
    return sb;
 }
 
+LibSampleTrack::BlockPtr SqliteSampleBlock::Get()
+{
+   assert(mSampleCount > 0);
+
+   // First try to acquire the cache without locking. If that doesn't succeed,
+   // lock and try again. Can't remember how this pattern is called.
+   auto cache = mCache.lock();
+   if (cache)
+      return cache;
+
+   std::lock_guard<std::mutex> lock(mCacheMutex);
+   cache = mCache.lock();
+   if (cache)
+      return cache;
+
+   // For debugging - will be gone.
```

Comment:
I wrote that debugging code to make sure that 
> cached data is released soon after last needed

Playback of a 20s mono track yielded for me
```
0 00000250D1D69EE0: Hello!
5315 00000250D1D68EE0: Hello!
5357 00000250D1D69EE0: Bye!
10765 00000250D1D68F60: Hello!
10808 00000250D1D68EE0: Bye!
16209 00000250D1D69A60: Hello!
16262 00000250D1D68F60: Bye!
20354 00000250D1D69A60: Bye!
```
Columns are `elapsed time (ms) / object address / ctor-dtor message` (See implementation of `LibSampleTrack::Block` for more detail).